### ISSUE TITLE
Exceptions while Building Search Indexes

### ISSUE THREAD

**Issue by GiovanniGabbolini** [metabrainz/musicbrainz-docker#140]
Hello,

I'm getting exceptions during the procedure that build the search indexes, i.e. running
`sudo docker-compose exec indexer python -m sir reindex`.

The exceptions happen while importing annotations:
`2020-04-28 13:19:48,873: Failed to import annotation with id in bounds (697237, 702246)
2020-04-28 13:19:48,880: (psycopg2.OperationalError) server closed the connection unexpectedly. This probably means the server terminated abnormally before or while processing the request.`

Moreover, I get exceptions building a single type of index, e.g. artist.

Running:
`sudo docker-compose exec indexer python -m sir reindex --entity-type artist`

Produces a number of similar exceptions:
`2020-04-28 19:20:25,553: Failed to import artist with id in bounds (1171391, 1176693)
2020-04-28 19:20:25,554: (psycopg2.OperationalError) server closed the connection unexpectedly.This probably means the server terminated abnormally before or while processing the request.
 [SQL: 'SELECT musicbrainz.area_alias.area AS musicbrainz_area_alias_area, musicbrainz.area_alias.type AS musicbrainz_area_alias_type, musicbrainz.area_alias.id AS musicbrainz_area_alias_id, musicbrainz.area_alias.name AS musicbrainz_area_alias_name, area_1.id AS area_1_id \nFROM (SELECT DISTINCT musicbrainz.artist.begin_area AS musicbrainz_artist_begin_area \nFROM musicbrainz.artist \nWHERE musicbrainz.artist.id >= %(id_1)s AND musicbrainz.artist.id < %(id_2)s) AS anon_1 JOIN musicbrainz.area AS area_1 ON area_1.id = anon_1.musicbrainz_artist_begin_area JOIN musicbrainz.area_alias ON area_1.id = musicbrainz.area_alias.area ORDER BY area_1.id'] [parameters: {'id_2': 1176693, 'id_1': 1171391}]`

together with other related errors:
`2020-04-28 19:28:16,579: Failed to import artist with id in bounds (1562266, 1567502)
2020-04-28 19:28:16,596: (psycopg2.OperationalError) FATAL:  the database system is in recovery mode`

Because of the multiprocessing, the stack traces are printed on screen not in sequence. However, those are the two kinds of exceptions I'm getting.

Version:
MusicBrainz Docker: 8dd3e5d
Docker Compose: 1.25.4
Docker Client/Server: 19.03.8/19.03.8

OS: MacOS 10.15.4
CPU: 8 threads
RAM: 8 gigs

**Comment by yvanzo** [comment]
Does `sudo docker-compose logs --timestamps db` provides more info?

**Comment by GiovanniGabbolini** [comment]
I ran again `sudo docker-compose exec indexer python -m sir reindex --entity-type artist`. 
The importing seems to be starting properly:

`Start reindexing artist...
2020-04-29 08:13:25,030: Checking whether the versions of the Solr cores are supported
2020-04-29 08:13:25,369: Importing artist...`

and it stays still for almost half an hour. During this phase, `sudo docker-compose logs --timestamps db` prints a number of:

`db_1           | 2020-04-29T08:14:14.515962800Z LOG:  unexpected EOF on client connection with an open transaction`

but the procedure continues without exceptions.
Then, an exception occurs:
[stack_trace.txt](https://github.com/metabrainz/musicbrainz-docker/files/4550660/stack_trace.txt)

This is the consequent output of `sudo docker-compose logs --timestamps db`:
[log_db.txt](https://github.com/metabrainz/musicbrainz-docker/files/4550661/log_db.txt)

**Comment by yvanzo** [comment]
Thank you for the details.

I have been able to partially reproduce this issue (without “recovery mode” message) and only once. The three next attempts succeeded to rebuild the `artist` search index with neither error nor warning. So I suggest you to try it again or to load search indexes dumps available from FTP instead.

I created the ticket SEARCH-610 to allow selecting search indexes to be updated live. It might be currently possible by deleting database triggers that are unneeded but I did not test it yet.
